### PR TITLE
Adds a harness for reward network

### PR DESCRIPTION
**Pull request by soumide1102**
This runs on chicoma. The training script includes DDP and LR scheduler. New functions have been added in torch_training_utils that supports training & validation for the reward network.

### CODE REVIEW

**Comment by hickmank** [comment]
@soumide1102 the coverage has to be above 60%. You're going to have to add tests to bring it back above 60%

File: src/yoke/torch_training_utils.py [hunk 1]
```diff
@@ -2285,3 +2285,204 @@ def train_lsc_policy_epoch(
...
+    # Rank 0 concatenates and saves or returns all losses
+    if rank == 0:
+        all_losses = torch.cat(gathered_losses, dim=0)  # Shape: (total_batch_size,)
+    else:
+        all_losses = None
+
+    return reward, pred, all_losses
+
+
+def train_lsc_reward_epoch(
```
Comment (hickmank) [inline src/yoke/torch_training_utils.py:2401]: You should make this accept a learning rate scheduler as well.

File: applications/harnesses/ch_lsc_reward/train_lsc_reward.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        # Train and Validate
+        tr.train_lsc_reward_epoch(
+            training_data=train_dataloader,
+            validation_data=val_dataloader,
+            num_train_batches=train_batches,
+            num_val_batches=val_batches,
+            model=model,
+            optimizer=optimizer,
+            loss_fn=loss_fn,
+            LRsched=LRsched,
```
Comment (hickmank) [inline applications/harnesses/ch_lsc_reward/train_lsc_reward.py:306]: I think this function currently doesn't accept the LRsched parameter.

File: applications/harnesses/ch_lsc_reward/START_study.py
Comment (hickmank) [inline applications/harnesses/ch_lsc_reward/START_study.py:1]: This should just be a symbolic link to applications/harnesses/START_study.py

File: src/yoke/torch_training_utils.py [hunk 3]
```diff
@@ -2285,3 +2285,204 @@ def train_lsc_policy_epoch(
...
+        for trainbatch_ID, traindata in enumerate(training_data):
+            # Stop when number of training batches is reached
+            if trainbatch_ID >= num_train_batches:
+                break
+
+            # Perform a single training step
+            reward_true, pred_mean, train_losses = train_lsc_reward_datastep(
+                traindata, model, optimizer, loss_fn, device, rank, world_size
+            )
+
```
Comment (hickmank) [inline src/yoke/torch_training_utils.py:2455]: Increment your LRsched scheduler in here.

**Review by hickmank — CHANGES_REQUESTED** [review]
You need to add the LRsched parameter to your epoch training function. 

Also, since we've finally gone below 60% coverage you have to write tests to bring coverage back up.

**Comment by soumide1102** [comment]
@hickmank if this looks okay to you, we can merge this.

I am doing some tests for fixing memory errors. I'll make a separate PR if any of this code changes after my tests.

**Comment by soumide1102** [comment]
Looks like I have to add some more tests. Coverage decreased to 59.636%. :(

**Review by hickmank — APPROVED** [review]
Awesome! Merge away!
